### PR TITLE
Polish codes for LocationComponentPlugin.createDefault2DPuck

### PR DESCRIPTION
**Pull request by Chaoba**
### Summary of changes
This PR polish the codes for https://github.com/mapbox/mapbox-maps-android/pull/1012#discussion_r782139029

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by kiryldz** [comment]
@Chaoba should it be rebased to v10.3 branch?

**Review by kiryldz — APPROVED** [review]

**Comment by Chaoba** [comment]
> @Chaoba should it be rebased to v10.3 branch?

@kiryldz It only polish codes, no need to cherry-pick into v10.3

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentExt.kt [hunk 1]
```diff
@@ -24,38 +30,16 @@ val MapPluginProviderDelegate.location: LocationComponentPlugin
...
-        R.drawable.mapbox_user_icon_shadow,
-        null
-      )
-    )
-  }
-}
\ No newline at end of file
+): LocationPuck2D = LocationPuck2D(
+  topImage = context.getCompatDrawable(R.drawable.mapbox_user_icon),
+  bearingImage = if (withBearing) context.getCompatDrawable(
```
Comment (yunikkk) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentExt.kt:35]: nit : 
```
context.getCompatDrawable(
if (withBearing)     
    R.drawable.mapbox_user_bearing_icon
else 
    R.drawable.mapbox_user_stroke_icon
  )
```

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentExt.kt [hunk 2]
```diff
@@ -15,6 +15,12 @@ val MapPluginProviderDelegate.location: LocationComponentPlugin
   @JvmName("getLocationComponent")
   get() = this.getPlugin(Plugin.MAPBOX_LOCATION_COMPONENT_PLUGIN_ID)!!
 
+private fun Context.getCompatDrawable(resId: Int) = ResourcesCompat.getDrawable(
```
Comment (yunikkk) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentExt.kt:18]: nit : add @resid annotation

**Review by yunikkk — APPROVED** [review]

**Pull request merged**